### PR TITLE
User instructions for installing Model Registry

### PR DESCRIPTION
**Pull request by rareddy**
This PR includes documentation for installing Model Registry using `oc` command on a terminal. This is especially designed for an end-user, not a developer.

## How Has This Been Tested?
This has been verified on OpenShift Cluster manually

## Merge criteria:
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by tarilabs — COMMENTED** [review]
thank you @rareddy I believe these are very helpful!

Some small considerations below:

File: docs/install.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+```
+oc patch odhdashboardconfig.opendatahub.io odh-dashboard-config -n opendatahub --type merge -p '{"spec": {"dashboardConfig": {"disableModelRegistry": false}}}'
+```
+## Find Model Registry URL
+Model Registry uses service mesh and opens a Gateway for external clients to reach. Execute following to find the URL where Model Registry is available.
+
+```
+URL=`echo "https://$(oc get routes -n istio-system -o json | jq '.items[2].status.ingress[0].host')" | tr -d '"'`
+```
```
[the remark this answers is not in the document]
Comment (Al-Pragliola) [reply docs/install.md:110]: you could use the label app name:

`oc get routes -n istio-system -l app.kubernetes.io/name=modelregistry-public -o json | jq '.items[].status.ingress[].host | select(contains("-rest"))' | tr -d '"'`

File: docs/install.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+## Install Database (skip if you using existing database)
+
+Model registry currently requires MySQL database 8.0.3 or above to function correctly. If you have a database already available you can skip this section all toghether. For "Development" or "NON-PRODUCTION" scenarios you can use following script to install MySQL database.
+
+```
+oc new-project test-database
+oc apply -f mysql-db.yaml
+oc get deployment model-registry-db    # make sure the database in Ready state
```
[the remark this answers is not in the document]
Comment (rareddy) [reply docs/install.md]: adopted the changes

File: docs/install.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+## install Model Registry
+
+To install Model Registry use the following script
+
+```
+oc project odh-model-registries
+oc apply -f registry.yaml
+oc get modelregistry modelregistry-public  # make sure the registry is available and running
```
[the remark this answers is not in the document]
Comment (rareddy) [reply docs/install.md]: adopted the changes

File: docs/install.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rareddy) [reply docs/install.md]: adopted the changes

File: docs/install.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+## Install Database (skip if you using existing database)
+
+Model registry currently requires MySQL database 8.0.3 or above to function correctly. If you have a database already available you can skip this section all toghether. For "Development" or "NON-PRODUCTION" scenarios you can use following script to install MySQL database.
+
+```
+oc new-project test-database
+oc apply -f mysql-db.yaml
+oc get deployment model-registry-db    # make sure the database in Ready state
+```
+
```
[the remark this answers is not in the document]
Comment (rareddy) [reply docs/install.md:385]: adopted the changes

[the remark this answers is not in the document]
Comment (rareddy) [reply docs/install.md:385]: adopted the changes

**Review by tarilabs — DISMISSED** [review]
Thank you @rareddy for applying feedback from myself and @Al-Pragliola in

and
/lgtm
overall.

The `docs/introduction.md` looks similar in part to the https://www.kubeflow.org/docs/components/model-registry/overview/#what-is-model-registry

Do you think valuable to reflect the other part in upstream too?

**Comment by rareddy** [comment]
Thank you for the feedback @tarilabs and @Al-Pragliola.

I have one more update to the usage of the model registry with UI which is still in progress, I will try to submit it soon.

>The docs/introduction.md looks similar in part to the https://www.kubeflow.org/docs/components/model-registry/overview/#what-is-model-registry

>Do you think valuable to reflect the other part in upstream too?

It is the same, as I copied from there. My intention is to make this guide equivalent to that of what is available Kubeflow upstream, but more aligned with midstream that can serve as upstream for downstream documentation, that we can feed to docs team. However, I am open to suggestions on how we can de-duplicate the information.

File: docs/install.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tarilabs) [reply docs/install.md:385]: I don't think this suggestion was applied.
re-opening. 

I recommend for this option to be shown to the user, let me know what you think?

[the remark this answers is not in the document]
Comment (rareddy) [reply docs/install.md:385]: Thanks for catching this @tarilabs 

I am also trying to see if we can 
* Move this documentation from this repo into a separate location such that is not tied to the operator documentation. 
* Also anyone can follow the documentation without downloading any files from repository.

Thus I did not select the above suggestion,  however failed to leave the correct comment by mistake.

[the remark this answers is not in the document]
Comment (tarilabs) [reply docs/install.md:385]: thanks for the clarifications; resolving this thread then.

File: docs/how-to-use-ui.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+![Model Registry](./images/dashboard-modelregistry-catalog1.png)
+
+it will bring this screen for registering a model
+
+![Model Registry](./images/dashboard-modelregistry-registermodel.png)
+
+Another way register the model is using [Python library](./getting-started.md). Note that when you register a model first time, it creates a model record and another record for a initial version that is defined. If you ever need to deploy a another iteration of the model after re-training try to add a new version of the model under the same original model record. So, a model record can contain any number of versions underneath. Each version is individually managed. The below screen shows when model is drilled down to version level.
```
Comment (tarilabs) [inline docs/how-to-use-ui.md:62]:
```suggestion
Another way register the model is using [Python library](./getting-started.md). Note that when you register a model the first time, it creates a Registered Model entry and related entries for the initial version that is defined. If you ever need to deploy another iteration of the model after re-training, try to add a new Model Version of the model under the original Registered Model entry. A Registered Model can contain any number of Model Versions underneath. Each version is individually managed. The below screen shows when model is drilled down to version level.
```

I would adopt the Logical Model jargon here, which is also reflected in the UI. wdyt?

**Review by tarilabs — COMMENTED** [review]
thank you for the UI variant, really appreciated.

/lgtm 
with one comment about jargon below:

**Comment by rareddy** [comment]
looks like somehow I screwed the merge of your changes @tarilabs, need to see how to revert and have a clean merge

**Comment by tarilabs** [comment]
> looks like somehow I screwed the merge of your changes @tarilabs, need to see how to revert and have a clean merge

We could always add the co-authored-by in the squash commit during merge if we wanted (noted for who press the button), anyhow no worries.

Anyhow in my view the most pressing thing is having this merged, so we could iterate progressively, rather than on the same PR my2c

**Comment by rareddy** [comment]
@tarilabs  thank you. agreed, I have another topic on "secure database configuration" I need to follow this PR. Not sure about the build issues, if this PR is good I would like to merge this.

**Comment by tarilabs** [comment]
> Not sure about the build issues

in this PR, everything is inside `/docs/**` and the GHAs:

https://github.com/opendatahub-io/model-registry-operator/blob/002ecbbdd95132b3b9ea9c4e06f660d652be9adc/.github/workflows/build.yml#L7-L16

https://github.com/opendatahub-io/model-registry-operator/blob/002ecbbdd95132b3b9ea9c4e06f660d652be9adc/.github/workflows/build-image-pr.yml#L4-L12

so I believe is just GH interface "confused".

**Pull request merged**